runtime/autoload/man.vim

```diff
@@ -1,137 +1,211 @@
+        \ 'buf': bufnr('%'),
+        \ 'lin': line('.'),
+        \ 'col': col('.')
+        \ })
+endfunction
 
-  let editcmd = 'edit'
-  " Use an existing 'man' window, else open a new one.
-  if &filetype !=# 'man'
+" find the closest man window above/left
+function! s:find_man(cmd) abort
+  if winnr('$') > 1
     let thiswin = winnr()
-    wincmd b
-    if winnr() > 1
-      exec thiswin . 'wincmd w'
-      while 1
-        if &filetype ==# 'man'
-          break
-        endif
-        wincmd w
-        if thiswin == winnr()
-          break
-        endif
-      endwhile
-    endif
+    while 1
+      if &filetype ==# 'man'
+        return 'edit'
+      endif
+      wincmd w
+      if thiswin == winnr() | break | endif
+    endwhile
+  endif
+  return a:cmd
+endfunction
 
-    if &filetype !=# 'man'
-      let editcmd = 'tabnew'
-    endif
+" parses the sect/page out of 'page(sect)'
+function! s:parse_page_and_section(fpage) abort
+  let ret = split(a:fpage, '(')
+  if len(ret) == 2 && ret[1] =~# '^\f\+)\f*$'
+    let iret = split(ret[1], ')')
+    return [ret[0], iret[0]]
+  elseif len(ret) == 1
+    return [ret[0], '']
+  else
+    return ['', '']
   endif
+endfunction
 
-  silent exec editcmd.' man://'.page.(empty(sect)?'':'('.sect.')')
+" returns the path of a manpage
+function! s:find_page(sect, page) abort
+  return split(system(s:man_cmd.' -w '.a:sect.' '.a:page), '\n')
+endfunction
 
-  setlocal modifiable
-  silent keepjumps norm! 1G"_dG
-  if empty($MANWIDTH)
-    let $MANWIDTH = winwidth(0)
+" parses the section out of the path to a manpage
+function! s:parse_sect(path) abort
+  let tail = fnamemodify(a:path, ':t')
+  if fnamemodify(tail, ":e") =~# '\%('.s:man_extensions.'\)\n'
+    let tail = fnamemodify(tail, ':r')
   endif
-  silent exec 'r!/usr/bin/man '.s:cmd(sect, page).' | col -b'
-  " Remove blank lines from top and bottom.
+  return substitute(tail, '\f\+\.\([^.]\+\)', '\1', '')
+endfunction
+
+function! s:read_page(sect, page, cmd)
+  silent execute a:cmd 'man://'.a:page.(empty(a:sect)?'':'('.a:sect.')')
+  setlocal modifiable
+  " remove all the text, incase we already loaded the manpage before
+  silent keepjumps normal! gg"_dG
+  let $MANWIDTH = winwidth(0)-1
+  " read manpage into buffer
+  silent execute 'r!'.s:man_cmd.' '.a:sect.' '.a:page
+  " remove all those backspaces
+  silent! keepjumps %substitute,.,,g
+  " remove blank lines from top and bottom.
   while getline(1) =~# '^\s*$'
     silent keepjumps 1delete _
   endwhile
   while getline('$') =~# '^\s*$'
     silent keepjumps $delete _
   endwhile
-  setlocal nomodified
+  keepjumps normal! gg
   setlocal filetype=man
-
-  if invoked_from_man || editcmd ==# 'tabnew'
-    call s:set_window_local_options()
-  endif
 endfunction
 
-function s:set_window_local_options() abort
-  setlocal colorcolumn=0 foldcolumn=0 nonumber
-  setlocal nolist norelativenumber nofoldenable
+function! s:error(msg) abort
+  redraws!
```

Comment:
wow, I didn't even know that was an abbreviation lol. I've never seen its full form.